src/ToolBox/SOS/Strike/util.cpp

```diff
@@ -6185,184 +5975,93 @@ HRESULT __stdcall PERvaMemoryReader::ReadExecutableAtRVA(DWORD relativeVirtualAd
+    // TODO: in-memory windows PDB not supported
+    hr = LoadSymbolsForWindows(pMD, moduleData.LoadedPEAddress, pModuleName, moduleData.bIsInMemory);
+    if (SUCCEEDED(hr))
     {
-        ExtErr("Error: Fail to initialize CoreCLR\n");
-        return Status;
+        return hr;
     }
+#endif // FEATURE_PAL
 
-    coreclr_create_delegate_ptr CreateDelegate =
-        (coreclr_create_delegate_ptr)dlsym(coreclrLib, "coreclr_create_delegate");
-
-    IfFailRet(CreateDelegate(hostHandle, domainId, SymbolReaderDllName,
-                        SymbolReaderClassName, "ResolveSequencePoint",
-                        (void **)&resolveSequencePointDelegate));
-    IfFailRet(CreateDelegate(hostHandle, domainId, SymbolReaderDllName,
-                        SymbolReaderClassName, "LoadSymbolsForModule",
-                        (void **)&loadSymbolsForModuleDelegate));
-    IfFailRet(CreateDelegate(hostHandle, domainId, SymbolReaderDllName,
-                        SymbolReaderClassName, "GetLocalVariableName",
-                             (void **)&getLocalVariableNameDelegate));
-    IfFailRet(CreateDelegate(hostHandle, domainId, SymbolReaderDllName,
-                        SymbolReaderClassName, "GetLineByILOffset",
-                             (void **)&getLineByILOffsetDelegate));
-    return Status;
+    return LoadSymbolsForPortablePDB(
+        moduleData.bIsInMemory,
+        pModuleName, 
+        moduleData.LoadedPEAddress,
+        moduleData.LoadedPESize, 
+        moduleData.InMemoryPdbAddress,
+        moduleData.InMemoryPdbSize);
 }
 
-HRESULT SymbolReader::GetLineByILOffset(__in_z const char* szModuleName, mdMethodDef MethodToken,
-                                         ULONG64 IlOffset, ___out ULONG *pLinenum,
-                                         __out_ecount(cbFileName) LPSTR lpszFileName,
-                                         ___in ULONG cbFileName)
-{
-    HRESULT Status = S_OK;
-
-    if (getLineByILOffsetDelegate == nullptr)
-    {
-        IfFailRet(SymbolReader::PrepareSymbolReader());
-    }
-
-    BSTR wszFileName = SysAllocStringLen(0, cbFileName);
-    if (wszFileName == nullptr)
-    {
-        return E_OUTOFMEMORY;
-    }
-
-    if (getLineByILOffsetDelegate(szModuleName, MethodToken, IlOffset, pLinenum, &wszFileName) == TRUE)
-    {
-        WideCharToMultiByte(CP_ACP, 0, wszFileName, (int) (_wcslen(wszFileName) + 1),
-                            lpszFileName, cbFileName, NULL, NULL);
-        if (*pLinenum == 0)
-        {
-            *pLinenum = -1;
-            Status = E_FAIL;
-        }
-    }
-    else
-    {
-        Status = E_FAIL;
-    }
-
-    SysFreeString(wszFileName);
-
-    return Status;
-}
-#endif //FEATURE_PAL
+#ifndef FEATURE_PAL
 
-HRESULT SymbolReader::LoadSymbols(IMetaDataImport * pMD, ULONG64 baseAddress, __in_z WCHAR* pModuleName, BOOL isInMemory)
+HRESULT SymbolReader::LoadSymbolsForWindows(IMetaDataImport* pMD, ULONG64 peAddress, __in_z WCHAR* pModuleName, BOOL isInMemory)
```

Comment:
Actually I think I meant originally was “LoadSymbolsForWindowsPDB”. How about that?